gobblin-compaction/src/main/java/gobblin/compaction/action/CompactionCompleteFileOperationAction.java

```diff
@@ -0,0 +1,31 @@
+package gobblin.compaction.action;
+
+import gobblin.compaction.mapreduce.CompactionAvroJobConfigurator;
+import gobblin.configuration.State;
+import gobblin.dataset.FileSystemDataset;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * A type of post action {@link CompactionCompleteAction} which focus on the file operations
+ */
+@Slf4j
+public class CompactionCompleteFileOperationAction extends CompactionCompleteAction<FileSystemDataset>{
+  private CompactionAvroJobConfigurator configurator;
+  public CompactionCompleteFileOperationAction(State state, CompactionAvroJobConfigurator configurator) {
+    super(state);
+    this.configurator = configurator;
+  }
+
+  /**
+   * Perform some file operations like move, rename, overwrite, delete, etc.
+   * @return status object or null depends on user implementation
+   */
+  public Object onCompactionJobCompelete (FileSystemDataset dataset) {
```

Comment:
Maybe I should put a void type here. The idea is to return any user defined type, but now I think it may not be needed